src/mpl2/src/hier_rtlmp.cpp

```diff
@@ -2545,22 +2537,18 @@ void HierRTLMP::calHardMacroClusterShape(Cluster* cluster)
   // To generate different macro tilings, we vary the outline constraints
   // we first vary the outline width while keeping outline_height fixed
   // Then we vary the outline height while keeping outline_width fixed
-  // We vary the outline of cluster to generate differnt tilings
+  // We vary the outline of cluster to generate different tilings
   std::vector<float> vary_factor_list{1.0};
   float vary_step = 1.0 / num_runs_;  // change the outline by at most halfly
   for (int i = 1; i < num_runs_; i++) {
     vary_factor_list.push_back(1.0 - i * vary_step);
   }
-  int run_thread = num_threads_;
   int remaining_runs = num_runs_;
   int run_id = 0;
   while (remaining_runs > 0) {
     std::vector<SACoreHardMacro*> sa_vector;
-    run_thread
-        = (remaining_runs > num_threads_) ? num_threads_ : remaining_runs;
-    if (graphics_) {
-      run_thread = 1;
-    }
+    const int run_thread
+        = graphics_ ? 1 : std::min(remaining_runs, num_threads_);
```

Comment:
Should this just be a function instead getThreadCount or something?

Reply:
remaining_runs is changing in the loop